plutus-list/src/PlutusTx/List/Ord.hs

```diff
@@ -0,0 +1,76 @@
+{-# LANGUAGE Trustworthy #-}
+
+{- |
+ Module: PlutusTx.List.Ord
+ Copyright: (C) MLabs 2021
+ License: Apache 2.0
+ Maintainer: Koz Ross <koz@mlabs.city>
+ Portability: GHC only
+ Stability: Experimental
+ Functions for checking the sortedness of lists.
+-}
+module PlutusTx.List.Ord (
+  isSorted,
+  isSortedOn,
+  isSortedMonotonic,
+  isSortedMonotonicOn,
+  isSortedBy,
+) where
+
+import PlutusTx.Prelude
+import Prelude qualified ()
+
+{-# INLINEABLE isSortedBy #-}
+
+{- | Checks if the 'Foldable' is sorted with respect to a total order
+represented by a binary relation.
+
+@since 1.0
+-}
+isSortedBy :: Foldable f => (a -> a -> Bool) -> f a -> Bool
```

Comment:
This is Boolean blind; a better choice for the function argument is `a -> a -> Ordering`.

Reply:
If we use `Ordering` then we can't implement `isSortedMonotonic(On)` in terms of this; we'll need to define another `isSortedMonotonicBy`. Not sure if this is good.

Also, I'm not sure if passing in a function returning `Ordering` is less blind. It actually can confuse user more b/c nowhere in the signature suggests if the function checks for ascending or descending order. By passing a `Bool`-returning function instead there isn't such confusion. For example, I'd argue it is more clear to write `isSortedBy (>=)` compared to ``isSortedBy (compare `on` Down)``.

Reply:
Fair point on both, let's leave it as-is then.